include/grpc/grpc.h

```diff
@@ -387,7 +386,8 @@ grpc_call *grpc_channel_create_registered_call(
    completion of type 'tag' to the completion queue bound to the call.
    The order of ops specified in the batch has no significance.
    Only one operation of each type can be active at once in any given
-   batch. */
+   batch. 
+   THREAD SAFETY: grpc_call_start_batch is thread-compatible */
```

Comment:
Ah, thanks for clarification. So what you meant is:
1. access call_start_batch needs to be synchronized if used from multiple threads 
2. rule 1 is valid for all batch operations with the exception of  read and write operations, that can be synchronized separately (as an optimization allowing fully independent reading vs writing)